File: MidiTeslaCoilApp/app/src/main/java/com/example/miditeslacoilapp/ui/InterruptorFragment.java
package com.example.miditeslacoilapp.ui;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.SeekBar;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;

import com.example.miditeslacoilapp.ViewModel.BluetoothViewModel;
import com.example.miditeslacoilapp.R;

import app.akexorcist.bluetotohspp.library.BluetoothSPP;

public class InterruptorFragment extends Fragment {
    private BluetoothSPP bluetooth;
    private static int frequencyThreshold = 60;
    private static int pwmThreshold = 100;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        BluetoothViewModel bluetoothViewModel = ViewModelProviders.of(getActivity()).get(BluetoothViewModel.class);
        bluetooth = bluetoothViewModel.getBluetoothSPP();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_interruptor, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        final TextView frequency = view.findViewById(R.id.freq);
        final TextView pulseWidth = view.findViewById(R.id.volume);

        SeekBar freqSeekBar = view.findViewById(R.id.freqSeekBar);
        SeekBar pwmSeekBar = view.findViewById(R.id.pwmSeekBar);

        freqSeekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser){
                Integer freq = seekBar.getProgress() + frequencyThreshold;
                frequency.setText("Frequency: " + freq.toString());
                bluetooth.send(freq.toString(), true);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {

            }
        });

        pwmSeekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                Integer pwm = seekBar.getProgress() + pwmThreshold;
                pulseWidth.setText("Pulse width: " + pwm.toString());
                bluetooth.send(pwm.toString() + "p", true);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {

            }
        });
    }

    @Override
    public void onPause() {
        super.onPause();
        bluetooth.send("0", true);
    }
}

File: MidiTeslaCoilApp/app/src/main/java/com/example/miditeslacoilapp/ui/InterruptorFragment.kt
package com.example.miditeslacoilapp.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.SeekBar
import android.widget.SeekBar.OnSeekBarChangeListener
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.fragment.app.activityViewModels
import com.example.miditeslacoilapp.R
import com.example.miditeslacoilapp.viewModels.BluetoothViewModel

class InterruptorFragment : Fragment() {
    private val viewModel: BluetoothViewModel by activityViewModels()

    companion object {
        private const val frequencyThreshold = 60
        private const val pwmThreshold = 100
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_interruptor, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val frequency = view.findViewById<TextView>(R.id.freq)
        val pulseWidth = view.findViewById<TextView>(R.id.volume)
        val freqSeekBar = view.findViewById<SeekBar>(R.id.freqSeekBar)
        val pwmSeekBar = view.findViewById<SeekBar>(R.id.pwmSeekBar)

        freqSeekBar.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                val freq = seekBar.progress + frequencyThreshold
                frequency.text = "Frequency: $freq"
                viewModel.writeData(freq.toString())
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })

        pwmSeekBar.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                val pwm = seekBar.progress + pwmThreshold
                pulseWidth.text = "Pulse width: $pwm"
                viewModel.writeData(pwm.toString())
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })
    }

    override fun onPause() {
        super.onPause()
        viewModel.onPause()
    }
}